### PR TITLE
Add workflow to check PRs to gh-pages

### PR DESCRIPTION
**Pull request by kdaily**
*Issue #, if available:*

NA

*Description of changes:*

Before allowing a merge, run `sphinx-build linkcheck` and `sphinx-lint`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/sphinx-check.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+        cache: pip
+    - name: Install dependencies
+      run: make install
+    - name: Run checks
+      run: make check
+    - name: Run link check
+      run: make linkcheck
```
Comment (kyleknap) [inline .github/workflows/sphinx-check.yml:22]: I mentioned this in a previous comment, but it would be nice if we had a single `make prcheck` rule that we could run locally and in CI so that we would not have to enumerate through the different make rule checks when making sure a PR will pass CI.

File: .github/workflows/sphinx-check.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps: 
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+        cache: pip
+    - name: Install dependencies
+      run: make install
+    - name: Run checks
```
Comment (kyleknap) [inline .github/workflows/sphinx-check.yml:19]: In addition to running the checks, we should also have a separate step that just builds the docs (i.e. `make html`) to make sure the documentation can be generated.

**Review by kyleknap — COMMENTED** [review]
Looks good. Just had a couple of suggestions.

File: .github/workflows/sphinx-check.yml (hunk 1 above)
Comment (kdaily) [reply .github/workflows/sphinx-check.yml:22]: 👍🏻 agree for the convenience on local side!

The idea for doing the steps separately in the CI was that I wanted to eventually have them show up on separate lines so it was more clear to the user which part the error was coming from, but I didn't finish investigating how to share steps between jobs so install and setup doesn't need to happen more than once.

Added it as a single step for now and can revisit.

**Comment by kdaily** [comment]
Confirmed it works with changes to workflow:

https://github.com/aws/aws-cli/actions/runs/2435117577

File: .github/workflows/sphinx-check.yml (hunk 1 above)
Comment (kdaily) [reply .github/workflows/sphinx-check.yml]: I guess it's not straightforward/worth the effort to reuse the install parts. Here's an example using a matrix strategy to run each make target, which I think makes it more clear what is being run. The `make prcheck` could still be used locally.

- https://github.com/kdaily/aws-cli/pull/28/files
- https://github.com/kdaily/aws-cli/pull/28

Comment (kyleknap) [reply .github/workflows/sphinx-check.yml]: With regards to command output, I agree the output is not necessarily easy to scan through. However, with regards to these checks, the user should not have to worry about the output in the first place; it should be something that they quickly run or have run in CI and only have to check the return code is zero and be able to move on quickly.

With regards to errors, is there are a specific error message that is hard to follow? With the way it is formatted now, if any of the checkers fail, it should fast fail meaning the failed command + traceback should just be at the bottom of the logs and you should not have to look through the logs of other commands for other failures.

**Review by kyleknap — APPROVED** [review]
Looks good! :ship:
